tests/test_remote.py

```diff
@@ -284,60 +322,118 @@ def quantum_function(theta, phi):
-                0.0,
-                0.0,
-                0.0,
-                0.1,
-                0.0,
-                0.1,
-                0.1,
-                0.1,
-                0.0,
-            ]
-        )
+        probs = quantum_function(1.0, 0)
+
+        # with dev.cutoff = 6, probs will be a 36-dimensional flat array. Since
+        # partial_probs is a 25-dimensional flat array, we first reshape to a 5x5 matrix and then
+        # pad to get a 6x6 matrix, finally flattening
+        exp_probs = np.reshape(partial_probs, (5, 5))
+        exp_probs = np.pad(exp_probs, [(0, 1), (0, 1)]).ravel()
+
+        assert probs.shape == (dev.cutoff ** len(wires),)
+        assert np.allclose(probs, exp_probs, atol=tol)
+
+    def test_probs_all_modes_high_cutoff(self, monkeypatch, tol):
+        """Tests that probabilities are correctly returned when using all modes and where
+        the device cutoff is above the samples cutoff."""
+        wires = range(8)
+        shots = 10
+        monkeypatch.setattr("strawberryfields.RemoteEngine", MockEngine)
+        dev = qml.device("strawberryfields.remote", backend="X8", shots=shots, cutoff_dim=6)
+
+        @qml.qnode(dev)
+        def quantum_function(theta, phi):
+            qml.Beamsplitter(theta, phi, wires=[0, 1])
+            qml.Beamsplitter(theta, phi, wires=[4, 5])
+            return qml.probs(wires=wires)
 
         probs = quantum_function(1.0, 0)
 
-        # all_fock_probs uses a cutoff equal to 1 + the maximum number of
-        # photons detected (including no photons detected)
-        cutoff = 1 + np.max(MOCK_SAMPLES)
+        exp_probs = np.reshape(full_probs, (5, 5, 5, 5, 5, 5, 5, 5))
+        exp_probs = np.pad(exp_probs, [(0, 1)] * 8).ravel()
+
+        assert probs.shape == (dev.cutoff ** len(wires),)
+        assert np.allclose(probs, exp_probs, atol=tol)
+
+    def test_probs_subset_modes_low_cutoff(self, monkeypatch, tol):
+        """Tests that probabilities are correctly returned when using a subset of modes and where
+        the device cutoff is below the samples cutoff."""
+        wires = [0, 1]
+        shots = 10
+        monkeypatch.setattr("strawberryfields.RemoteEngine", MockEngine)
+        dev = qml.device("strawberryfields.remote", backend="X8", shots=shots, cutoff_dim=3)
+
+        @qml.qnode(dev)
+        def quantum_function(theta, phi):
+            qml.Beamsplitter(theta, phi, wires=[0, 1])
+            qml.Beamsplitter(theta, phi, wires=[4, 5])
+            return qml.probs(wires=wires)
+
+        with pytest.warns(UserWarning, match="Samples were generated where at least one mode"):
+            probs = quantum_function(1.0, 0)
+
+        # with dev.cutoff = 3, probs will be a 9-dimensional flat array. Since
+        # partial_probs is a 25-dimensional flat array, we first reshape to a 5x5 matrix and then
+        # take the first 3 elements on both axes, finally flattening
+        exp_probs = partial_probs.reshape((5, 5))[:3, :3]
```

Comment:
Just double checking, but re-normalization does _not_ happen, right? I think that is the right call, since unnormalized probabilities can act as a warning to the user (in addition to the actual warning 😆 ) that their cutoff was too small.

Reply:
> But for me this is worse.

I agree! I like the current approach.